eth_contracts/contracts/BattleGroups.sol

```diff
@@ -11,16 +11,16 @@ contract BattleGroups {
 		address owner;
 
 		// Cards which belong to the BattleGroup
-		uint256[MAX_CARDS_PER_GROUP] cards;
+		uint256[5] cards;
```

Comment:
Why change this? The uint was a constant, however if web3js doesn't play well with variable length arrays, we might want to consider making battlegroups mutable.

Reply:
compilation fails

Reply:
Hm, it doesn't on my machine, but I'll have to look into it tomorrow!

Reply:
ok technically it works under truffle and all, but when you run `abigen` on the contract it breaks, I can attempt to repro later.

Reply:
@harrischristiansen this is the error

```
codegen_tools_abigen_1     | Failed to build Solidity contract: solc: exit status 1
codegen_tools_abigen_1     | /eth_contracts/contracts/BattleGroups.sol:14:11: Error: Invalid array length, expected integer literal.
codegen_tools_abigen_1     | 		uint256[MAX_CARDS_PER_GROUP] cards;
codegen_tools_abigen_1     | 		        ^-----------------^
```

so the solidity compiler doesn't like it, but truffle is OK with it 

thoughts @Shonas301 ?

https://ethereum.stackexchange.com/questions/12433/invalid-array-length-expected-integer-literal

Reply:
I mean, we need abigen approval at the end of the day and the tool is finicky as is, I'll create an issue, we can re-approach the problem when we look to deploy.